src/app/pages/payment/international-payment/international-payment.component.ts

```diff
@@ -0,0 +1,152 @@
+    private toastrService: ToastrService,
+    private location: Location,
+    private fb: FormBuilder,
+    private spinner: NgxSpinnerService,
+    private sanitizer: DomSanitizer
+
+  ) { }
+
+  generateYears() {
+    const thisYear = new Date().getFullYear();
+    this.expectedYears = Array.from(Array(14).keys()).map((item) => item + thisYear);
+  }
+
+
+  ngOnInit() {
+
+    this.generateYears();
+    this.form = this.fb.group({
+      cardNo: [
+        null,
+        [
+          Validators.required,
+          Validators.pattern('^[1-9]+[0-9]*$'),
+          Validators.minLength(8)
+        ]
+      ],
+      cvv: [
+        null,
+        [
+          Validators.required,
+          Validators.pattern('^[1-9]+[0-9]*$'),
+          Validators.maxLength(4),
+          Validators.minLength(3)
+        ]
+      ],
+      expiryMonth: [
+        null,
+        [Validators.required, Validators.max(12), Validators.min(1)]
+      ],
+      expiryYear: [null, [Validators.required, Validators.min(19)]],
+      amount: [
+        null,
+        [Validators.required, Validators.pattern('^[1-9]+[0-9]*$')]
+      ],
+      billingZip: [null, [Validators.required, Validators.pattern('^[1-9]+[0-9]*$')]],
+      billingCity: [null, [Validators.required]],
+      billingAddress: [null, [Validators.required]],
+      billingState: [null, [Validators.required]],
+      billingCountry: [null, [Validators.required]],
+      saveCard: false,
+      purpose: [null, [Validators.required]],
+    });
+  }
+
+  ngOnDestroy() {
+    removeSubscription(this.subscription);
+  }
+
+
+  backClicked() {
+    this.location.back();
+  }
+
+  onSubmitPaymentDetails({ value }) {
+
+    this.loading = true;
+    this.spinner.show();
+    this.payload = {
+      cardno: value.cardNo,
+      cvv: value.cvv,
+      expirymonth: value.expiryMonth,
+      expiryyear: value.expiryYear.slice(-2),
+      amount: value.amount,
+      billingzip: value.billingZip,
+      billingcity: value.billingCity,
+      billingaddress: value.billingAddress,
+      billingstate: value.billingState,
+      billingcountry: value.billingCountry,
+      save_card: value.saveCard,
+      purpose: value.purpose,
+    };
+
+    value.propertyId ? this.payload.property_id = this.propertyId : null;
+
+
+    this.subscription.push(
+      this.internationalPaymentService
+        .createInternationalPayment(this.payload)
+        .subscribe(
+          data => {
+            this.spinner.hide();
+            this.loading = false;
+            this.toastrService.success('Your transaction has been initiated. '
+              + 'Please ensure to enter the OTP sent to your phone');
+            this.iframeSrc = this.sanitizer.bypassSecurityTrustResourceUrl(data.message + '&output=embed');
+
+            this.shouldHide = false;
+          },
+          error => {
+            this.spinner.hide();
+            this.loading = false;
+
+            let errors = '';
+            for (const key in error) {
```

Comment:
`Ts-lint ` is complaining here.
![image](https://user-images.githubusercontent.com/12128153/63682688-269f3000-c801-11e9-8dfd-6a60bc7dea70.png)

What do you think of using `object.keys` or `object.entries` here. so that you can iterate through the error object with `.map`

https://appdividend.com/2019/03/15/javascript-object-values-example-object-values-tutorial/

Reply:
All done @Tevinthuku